Content.Server/Body/Systems/BloodstreamSystem.cs

```diff
@@ -120,8 +120,9 @@ public override void Update(float frameTime)
             if (!_solutionContainerSystem.ResolveSolution(uid, bloodstream.BloodSolutionName, ref bloodstream.BloodSolution, out var bloodSolution))
                 continue;
 
-            // Adds blood to their blood level if it is below the maximum; Blood regeneration. Must be alive.
-            if (bloodSolution.Volume < bloodSolution.MaxVolume && !_mobStateSystem.IsDead(uid))
+            var bloodPercentage = GetBloodLevelPercentage(uid, bloodstream);
+            // Adds blood to their blood level if it is below the reference; Blood regeneration. Must be alive.
+            if (bloodPercentage < 1 && !_mobStateSystem.IsDead(uid))
             {
                 TryModifyBloodLevel(uid, bloodstream.BloodRefreshAmount, bloodstream);
             }
```

Comment:
This could fill a bit above `BloodReferenceVolume` if you're missing less blood than `BloodRefreshAmount`. It should probably be capped.